cmd/ncproxy/ncproxy.go

```diff
@@ -557,6 +558,58 @@ func (s *grpcService) GetNetworks(ctx context.Context, req *ncproxygrpc.GetNetwo
 // client connections.
 type ttrpcService struct {
 	m sync.Mutex
+	// database for containerID to compute agent address
+	agentStore *computeAgentStore
+}
+
+func newTTRPCService(ctx context.Context, db *bolt.DB) *ttrpcService {
+	agentStore := newComputeAgentStore(db)
+	return &ttrpcService{
+		agentStore: agentStore,
+	}
+}
+
+func getComputeAgentClient(agentAddr string) (computeagent.ComputeAgentService, error) {
+	conn, err := winio.DialPipe(agentAddr, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to connect to compute agent service")
+	}
+	client := ttrpc.NewClient(
+		conn,
+		ttrpc.WithUnaryClientInterceptor(octtrpc.ClientInterceptor()),
+		ttrpc.WithOnClose(func() { conn.Close() }),
+	)
+	return computeagent.NewComputeAgentClient(client), nil
+}
+
+func (s *ttrpcService) reconnectComputeAgents(ctx context.Context) {
+	contents, err := s.agentStore.getActiveComputeAgents(ctx)
+	if err != nil && errors.Is(err, errBucketNotFound) {
+		// no entries in the database yet, return early
+		log.G(ctx).WithError(err).Debug("no entries in database")
+		return
+	} else if err != nil {
+		log.G(ctx).WithError(err).Warn("failed to get compute agent information")
+	}
+
+	for cid, addr := range contents {
+		service, err := getComputeAgentClient(addr)
+		if err != nil {
+			// can't connect to compute agent, remove entry in database
+			log.G(ctx).WithField("agent address", addr).WithError(err).Warn("failed to create new compute agent client")
+			dErr := s.agentStore.deleteComputeAgent(ctx, cid)
+			if dErr != nil {
+				log.G(ctx).WithField("key", cid).WithError(dErr).Warn("failed to delete key from compute agent store")
+			}
+			continue
+		}
+		log.G(ctx).WithField("container ID", cid).Info("reconnected to container's compute agent")
```

Comment:
Usually we don't use whitespace in logrus field names. This makes it a bit easier to query them in e.g. Kusto.